contracts/adapters/UniswapAdapter.sol

```diff
@@ -0,0 +1,209 @@
+        address _ethAddress,
+        address _wbtcDiggUniswap,
+        address _wbtcDiggSushiswap,
+        address _diggAddress
+    ) public {
+
+        wbtcAddress = _wbtcAddress;
+        ethAddress = _ethAddress;
+        diggAddress = _diggAddress;
+        uniswapRouter = IUniswapV2Router02(_uniswapRouter);
+        sushiswapRouter = IUniswapV2Router02(_sushiswapRouter);
+        wbtcDiggUniswap = IUniswapV2Pair(_wbtcDiggUniswap);
+        wbtcDiggSushiswap = IUniswapV2Pair(_wbtcDiggSushiswap);
+        wbtcToken = IERC20(_wbtcAddress);
+
+        SafeERC20.safeApprove(IERC20(_wbtcAddress), address(_uniswapRouter), type(uint256).max);
+        SafeERC20.safeApprove(IERC20(_wbtcAddress), address(_sushiswapRouter), type(uint256).max);
+        SafeERC20.safeApprove(IERC20(_diggAddress), address(_uniswapRouter), type(uint256).max);
+        SafeERC20.safeApprove(IERC20(_diggAddress), address(_sushiswapRouter), type(uint256).max);
+
+     }
+
+    receive() external payable {}
+
+    function protocolName() public pure returns (string memory) {
+        return _name;
+    }
+
+    function nonFungible() external pure  returns (bool) {
+        return _nonFungible;
+    }
+
+    function sqrt(uint y) internal pure returns (uint z) {
+        if (y > 3) {
+            z = y;
+            uint x = y / 2 + 1;
+            while (x < z) {
+                z = x;
+                x = (y / x + x) / 2;
+            }
+        } else if (y != 0) {
+            z = 1;
+        }
+    }
+
+    // Code from Alpha Homora
+    // The core math involved in getting optimal swap amt to provide amm liquidity
+    function getSwapAmt(uint256 amtA, uint256 resA) internal pure returns (uint256){
+        return sqrt(amtA.mul(resA.mul(3988000) + amtA.mul(3988009))).sub(amtA.mul(1997))/1994;
+    }
+
+    function validateExchange(string memory exchangeName) internal view returns (Exchange){
+        if (keccak256(abi.encodePacked(exchangeName)) == uniswapHash){
+                return Exchange.Uniswap;
+        }
+        else if (keccak256(abi.encodePacked(exchangeName)) == sushiswapHash){
+                return Exchange.Sushiswap;
+        }
+        require(false, 'invalid exchange');
+    }
+
+    function expectedWbtcOut(uint256 ethAmt, string memory exchangeName) public view returns (uint256){
+        Exchange exchange = validateExchange(exchangeName);
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        address[] memory path = new address[](2);
+        path[0] = wethAddress;
+        path[1] = wbtcAddress;
+        uint256 wbtcOut = router.getAmountsOut(ethAmt, path)[1];
+        return wbtcOut;
+    }
+
+    //this function returns both the expected digg amount out as well as the input trade amt of wbtc used
+    //these are both needed as inputs to buyLp
+    function expectedDiggOut(uint256 wbtcAmt, string memory exchangeName) public view returns (uint256[2] memory){

```

Comment:
Prefer to just return multiple uint's instead of an array:

```solidity
function expectedDiggOut(uint256 wbtcAmt, string memory exchangeName) public view returns (uint256 diggOut, uint tradeAmt){
```